packages/core/src/tile/model/tile.ts

```diff
@@ -1,26 +1,147 @@
+   */
+  PENDING_LOAD = 0b1,
+
+  /**
+   * The tile request has been dispatched.
+   */
+  LOADING = 0b10,
+
+  /**
+   * Tile has loaded and is currently being rendered.
+   */
+  VISIBLE = 0b100,
+
+  /**
+   * Tile has been marked for delete during the next cleanup step.
+   */
+  PENDING_DELETE = 0b1000
 }
 
-export abstract class Tile<T extends BaseResponse> {
-  protected mesh: Mesh;
-  protected scene: Scene;
+export class Tile<T, C extends TileContent = TileContent> {
+  /**
+   * Global tile counter used as unique id for tiles.
+   */
+  private static tileCount = 0;
 
-  public readonly index: number[];
+  /**
+   * Unique autogenerated tile id.
+   */
+  public readonly id: number = Tile.tileCount++;
 
-  constructor(scene: Scene, response: T, uniqueID?: string) {
-    this.index = response.index;
-    this.scene = scene;
-    this.mesh = new Mesh(`${this.index.toString()}_${uniqueID}`, this.scene);
-    this.mesh.alwaysSelectAsActiveMesh = true;
-  }
+  /**
+   * Unique tile identifier inside the dataset. For datasets with morton indexes
+   * this array contains only a single value, the morton index.
+   */
+  public index: number[];
+
+  /**
+   * The bounding info in the projected coordinate system.
+   */
+  public boundingInfo: BoundingInfo;
+
+  /**
+   * The tiles refinement strategy.
+   */
+  public refineStrategy: RefineStrategy;
+
+  /**
+   * The geometric error in the projected coordinate system.
+   */
+  public geometricError: number;
+
+  /**
+   * The child tiles of the tile. In case of implicit tiles more child tiles may be added during runtime.
+   */
+  public children: Array<Tile<T, C>>;
+
+  /**
+   * THe parent of the tile or undefined if the tile is the tileset root.
+   */
+  public parent?: Tile<T, C>;
 
-  public abstract update<U extends UpdateOptions<T>>(updateOptions: U): void;
+  /**
+   * The tile content.
+   */
+  public content: Array<T>;
+
+  /**
+   * The tiles tilling scheme. If value is `TillingScheme.NONE` indication of explicit tileset.
+   * Default value is `TillingScheme.NONE`.
+   */
+  public tillingScheme: TillingScheme;
+
+  /**
+   * The tile state the provides info regarding the loading and visibility state of the tile.
+   */
+  public state: TileState;
+
+  /**
+   *
+   */
+  public mask: number;
+
+  /**
+   * The tiles renderable data object used for updating the data state
+   */
+  public data?: C;
+
+  constructor() {
+    this.boundingInfo = new BoundingInfo(
+      Vector3.ZeroReadOnly,
+      Vector3.ZeroReadOnly
+    );
+    this.refineStrategy = RefineStrategy.ADD;
+    this.children = [];
+    this.content = [];
+    this.index = [];
+    this.parent = undefined;
+    this.tillingScheme = TillingScheme.NONE;
+    this.geometricError = 0;
+    this.state = TileState.HIDDEN;
+    this.mask = 0;
+  }
 
   public dispose() {
-    this.mesh.dispose(false, true);
+    this.data?.dispose();
+    this.data = undefined;
+  }
+
+  public screenSpaceError(camera: Camera): number {
```

Comment:
a sentence or three about where these math formulas come from, would be helpful for the general reader